packages/slice-machine/components/ScreenshotChangesModal/index.tsx

```diff
@@ -0,0 +1,205 @@
+import Card from "@components/Card";
+import { AiOutlinePicture } from "react-icons/ai";
+import { RiErrorWarningLine } from "react-icons/ri";
+import SliceMachineModal from "@components/SliceMachineModal";
+
+import { useSelector } from "react-redux";
+import { isModalOpen } from "@src/modules/modal";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { ModalKeysEnum } from "@src/modules/modal/types";
+
+import { Flex, Heading, Close, Box, Text } from "theme-ui";
+
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { useState } from "react";
+
+const VariationIcon: React.FC<{ isValid?: boolean }> = ({ isValid }) => (
+  <Flex
+    sx={{
+      p: 2,
+      border: "1px solid #E4E2E4",
+      borderRadius: "6px",
+      bg: "#FFF",
+    }}
+  >
+    {isValid ? (
+      <AiOutlinePicture style={{ fontSize: "16px" }} />
+    ) : (
+      <RiErrorWarningLine style={{ fontSize: "16px", color: "#ED811C" }} />
+    )}
+  </Flex>
+);
+
+const VariationsList = ({
+  slice,
+  onSelectVariation,
+  selectedSliceVariation,
+}: {
+  slice: ComponentUI;
+  isSelected?: boolean;
+  selectedSliceVariation: [string, string];
+  onSelectVariation: (sliceID: string, variationID: string) => void;
+}) => (
+  <>
+    <Heading
+      as="h4"
+      sx={{
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        textOverflow: "ellipsis",
+      }}
+    >
+      {slice.model.name}
+    </Heading>
+    <Box>
+      <Box
+        as="ul"
+        sx={{
+          pl: 0,
+          my: 3,
+        }}
+      >
+        {slice.model.variations.map((variation) => {
+          const isSelectedVariation =
+            selectedSliceVariation[0] === slice.model.id &&
+            selectedSliceVariation[1] === variation.id;
+          return (
+            <Box
+              as="li"
+              sx={{
+                listStyle: "none",
+                my: 2,
+                p: 2,
+                cursor: "pointer",
+                borderRadius: "4px",
+                ...(isSelectedVariation && {
+                  bg: "#EDECEE",
+                  fontWeight: "500",
+                }),
+                ":hover": {
+                  bg: "#EDECEE",
+                },
+              }}
+              onClick={() => onSelectVariation(slice.model.id, variation.id)}
+            >
+              <Flex
+                sx={{
+                  alignItems: "center",
+                }}
+              >
+                <VariationIcon isValid={!!slice.screenshots[variation.id]} />
+                <Text sx={{ ml: 2 }}>{variation.name}</Text>
+              </Flex>
+            </Box>
+          );
+        })}
+      </Box>
+    </Box>
+  </>
+);
+
+const ScreenshotChangesModal = ({
+  slices,
```

Comment:
It shouldn't take any parameters, they should come from the payload of the action.

Reply:
Can you show me how you would implement the modal in a gist or something?

Reply:
You can refer to my comment below but you’ll need a way to provide data to your modal, this can’t happen without props

Reply:
I suggest a way to simplify the implementation with tools like Radix but I’d suggest you @Heyrows that if you want to keep it homemade, you should maybe provide some insights on how you would build it. You have very different approach in mind and I think you should weight in and provide a concrete example that you can then discuss about together to reach consensus.